API-Samples/C#/ArmBased/VideoIndexerClient/VideoIndexerClient.cs

```diff
@@ -0,0 +1,318 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Threading.Tasks;
+using VideoIndexingARMAccounts.VideoIndexerClient.Auth;
+using VideoIndexingARMAccounts.VideoIndexerClient.Model;
+using VideoIndexingARMAccounts.VideoIndexerClient.Utils;
+using static VideoIndexingARMAccounts.Consts;
+
+namespace VideoIndexingARMAccounts.VideoIndexerClient
+{
+    public class VideoIndexerClient
+    {
+        private HttpClient _httpClient;
+        private string _armAccessToken;
+        private string _accountAccessToken;
+        private Account _account;
+
+        private readonly TimeSpan _pollingInteval = TimeSpan.FromSeconds(10);
+
+        public VideoIndexerClient()
+        {
+            System.Net.ServicePointManager.SecurityProtocol |= System.Net.SecurityProtocolType.Tls12 | System.Net.SecurityProtocolType.Tls13;
+            
+        }
+
+        public async Task Authenticate()
+        {
+            try
+            {
+                _armAccessToken = await AccountTokenProvider.GetArmAccessTokenAsync();
+                _accountAccessToken = await AccountTokenProvider.GetAccountAccessTokenAsync(_armAccessToken);
+                _httpClient = HttpClientUtils.CreateHttpClient();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Get Information about the Account
+        /// </summary>
+        /// <param name="accountName"></param>
+        /// <returns></returns>
+        public async Task<Account> GetAccount(string accountName)
+        {
+            if (_account != null)
+            {
+                return _account;
+            }
+            Console.WriteLine($"Getting account {accountName}.");
+            try
+            {
+                // Set request uri
+                var requestUri = $"{AzureResourceManager}/subscriptions/{SubscriptionId}/resourcegroups/{ResourceGroup}/providers/Microsoft.VideoIndexer/accounts/{accountName}?api-version={ApiVersion}";
+                var client = new HttpClient(new HttpClientHandler());

```

Comment:
Also here, you don't need to create HttpClient. What you can do is path the Authorization in the request and not in the default header. 

Also, a better approach is use local cache for the header and use it per request to ensure it will be refreshed before expired.

Reply:
I tried that . you have to pass the token in the header when interacting with VI ARM

Reply:
local cache and auto refresh - next PR

Reply:
re-creating the Http Client is in order to ensure that the DEFAULT Bearer token that was used for GetAccountAccessToken,
the other calls beyond it do not need this header .